File: ReservationManagement/src/main/java/com/example/reservationmanagement/model/Reservation.java
package com.example.reservationmanagement.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Reservation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer reservationID;

    private LocalDateTime DateIssued;
    private LocalDate reservationStartDate;
    private LocalDate reservationEndDate;
    private LocalTime startTime;
    private LocalTime endTime;
    private Integer duration;

    @Enumerated(EnumType.STRING)
    private ReservationStatus state;

    private Boolean deleted = Boolean.FALSE;
    private Integer cost;

    private Integer userId;
    private String hotelId;
    private String roomId;

    private LocalDateTime cancelledAt;
    private LocalDateTime deletedAt;

}

File: frontend/src/app/admin/models/reservation.ts
export type ReservationStatus = 'Confirmed' | 'Cancelled' | 'Completed';

export interface Reservation {
  reservationID: number;
  DateIssued: string;          // ISO string (LocalDateTime)
  reservationStartDate: string; // ISO date string (LocalDate)
  reservationEndDate: string;   // ISO date string (LocalDate)
  startTime: string;            // HH:mm:ss (LocalTime)
  endTime: string;              // HH:mm:ss (LocalTime)
  duration: number;
  cost: number;
  state: ReservationStatus;
  userId: number;
  hotelId: string;
  roomId: string;
}
